scipy/stats/_continuous_distns.py

```diff
@@ -4225,6 +4225,61 @@ def _stats(self, a, moments='mvsk'):
 
 skewnorm = skew_norm_gen(name='skewnorm')
 
+
+class trap_gen(rv_continuous):
+    """A trapezoidal continuous random variable.
+
+    %(before_notes)s
+
+    Notes
+    -----
+    The trapezoidal distribution can be represented with an up-sloping line
+    from ``loc`` to ``(loc + c*scale)``, then constant to ``(loc + d*scale)``
+    and then downsloping from ``(loc + d*scale)`` to ``(loc+scale)``.
+
+    `trap` takes ``c`` and ``d`` as shape parameters.
+
+    %(after_notes)s
+
+    The standard form is in the range [0, 1] with c the mode.
+    The location parameter shifts the start to `loc`.
+    The scale parameter changes the width from 1 to `scale`.
+
+    %(example)s
+
+    """
+    def _rvs(self, c, d):
+        rands = self._random_state.rand(self._size)
+        return self._ppf(rands, c, d)
```

Comment:
This is inherited from superclass,  not needed here https://github.com/scipy/scipy/blob/master/scipy/stats/_distn_infrastructure.py#L801